test/cri-containerd/gmsa.go

```diff
@@ -0,0 +1,48 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+// Generates a gmsa credential spec and places it at 'path'. Dir has to exist, it will not be created by
+// New-CredentialSpec.
+func generateCredSpec(path string) error {
+	output, err := exec.Command(
+		"powershell",
+		"New-CredentialSpec",
+		"-AccountName",
+		gmsaAccount,
+		"-Path",
+		path,
+	).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to create new credential spec: %s", string(output))
+	}
+	return nil
+}
+
+// Tries to generate a cred spec to use for gmsa test cases. Returns the cred
+// spec and an error if any.
+func gmsaSetup() (string, error) {
```

Comment:
With test utility functions it is nice to make them take a `t *testing.T` and just call `t.Fatal` on errors instead of returning. This lets us keep the actual test code shorter and more to the point.